client/webserver/site/src/js/markets.js

```diff
@@ -988,6 +969,181 @@ export default class MarketsPage extends BasePage {
   }
 }
 
+/*
+ * tmplElement is a helper function for grabbing sub-elements of the market list
+ * template.
+ */
+function tmplElement (ancestor, s) {
+  return ancestor.querySelector(`[data-tmpl="${s}"]`)
+}
+
+/*
+ * MarketList respresents the list of exchanges and markets on the left side of
+ * markets view. The MarketList provides utilities for adjusting the visibility
+ * and sort order of markets.
+ */
+class MarketList {
+  constructor (col) {
+    this.column = col
+    this.selected = null
+    const tmpl = tmplElement(col, 'xc')
+    cleanTemplates(tmpl)
+    this.xcSections = []
+    for (const dex of Object.values(app.user.exchanges)) {
+      this.xcSections.push(new ExchangeSection(tmpl, dex))
+    }
+    // Initial sort is alphabetical.
+    for (const xc of this.sortedSections()) {
+      this.column.appendChild(xc.box)
+    }
+  }
+
+  /*
+   * sortedSections returns a list of ExchangeSection sorted alphabetically by
+   * host.
+   */
+  sortedSections () {
+    return [...this.xcSections].sort((a, b) => a.host < b.host ? -1 : 1)
+  }
+
+  /*
+   * xcSection is a getter for the ExchangeSection for a specified host.
+   */
+  xcSection (host) {
+    for (const xc of this.xcSections) {
+      if (xc.host === host) return xc
+    }
+    return null
+  }
+
+  /* exists will be true if the specified market exists. */
+  exists (host, baseID, quoteID) {
+    const xc = this.xcSection(host)
+    if (!xc) return false
+    for (const mkt of xc.marketRows) {
+      if (mkt.baseID === baseID && mkt.quoteID === quoteID) return true
+    }
+    return false
+  }
+
+  /* first gets the first market from the first exchange, alphabetically. */
+  first () {
+    const firstXC = this.sortedSections()[0]
+    const firstMkt = firstXC.first()
+    return makeMarket(firstXC.host, firstMkt.baseID, firstMkt.quoteID)
+  }
+
+  /* select sets the specified market as selected. */
+  select (host, baseID, quoteID) {
+    if (this.selected) this.selected.row.classList.remove('selected')
+    this.selected = this.xcSection(host).marketRow(baseID, quoteID)
+    this.selected.row.classList.add('selected')
+  }
+
+  /* setConnectionStatus sets the visiblity of the disconnected icon based
+   * on the core.EpochNotification.
+   */
+  setConnectionStatus (note) {
+    this.xcSection(note.host).setConnected(note.connected)
+  }
+
+  /*
+   * setFilter sets the visibility of market rows based on the provided filter.
+   */
+  setFilter (filter) {
+    for (const xc of this.xcSections) {
+      xc.setFilter(filter)
+    }
+  }
+}
+
+/*
+ * ExchangeSection is a top level section of the MarketList.
+ */
+class ExchangeSection {
+  constructor (tmpl, dex) {
+    this.dex = dex
+    this.host = dex.host
+    const box = tmpl.cloneNode(true)
+    this.box = box
+    const header = tmplElement(box, 'header')
+    this.connectedIcon = tmplElement(header, 'disconnected')
+    if (dex.connected) Doc.hide(this.connectedIcon)
```

Comment:
`connectedIcon` -> `disconnectedIcon`
Wondered for a minute why we're hiding the `connectedIcon` if the dex is connected, before seeing `tmplElement(header, 'disconnected')`

Reply:
I was ok with seeing it all the time, but the absence of an icon for connected with just a disconnected icon was the intended behavior.  Keeping clutter off the UI is a good idea though.